src/Nethermind/Nethermind.Synchronization/BeamSync/BeamBlockchainProcessor.cs

```diff
@@ -260,18 +261,26 @@ private void BeamProcess(Block block)
                 _recoveryStep.RecoverData(block);
                 (IBlockchainProcessor beamProcessor, IStateReader stateReader) = CreateProcessor(block, new ReadOnlyDbProvider(_readOnlyDbProvider, true), _specProvider, _logManager);
 
-                BlockHeader parentHeader = _readOnlyBlockTree.FindHeader(block.ParentHash, BlockTreeLookupOptions.TotalDifficultyNotNeeded);
+                BlockHeader? parentHeader = _readOnlyBlockTree.FindHeader(block.ParentHash, BlockTreeLookupOptions.TotalDifficultyNotNeeded);
```

Comment:
nullability on Nethermind.Blockchain project should take aroun 6 hours to do and may reuire some more thinking here and there, it is just one of changes that will cause domino effects everywhere in the solution